llvm/include/llvm-c/cas-plugin.h

```diff
@@ -0,0 +1,38 @@
+/*===-- llvm-c/cas-plugin.h - CASPlugin Public C Interface --------*- C -*-===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* This header provides public interface to an abstract link time optimization*|
+|* library.  LLVM provides an implementation of this interface for use with   *|
+|* llvm bitcode files.                                                        *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_CAS_PLUGIN_H
+#define LLVM_C_CAS_PLUGIN_H
+
+#include "llvm-c/ExternC.h"
+
+/** opaque reference to a CASDB */
+typedef struct LLVMOpaqueCASDB *casdb_t;
+
+LLVM_C_EXTERN_C_BEGIN
+
+/**
+ * Create a plugin CAS with arguments. Return NULL if an error occers.
+ */
+extern casdb_t llvm_create_plugin_cas(int argc, const char *argv[]);
```

Comment:
Seems like we should specify in the documentation that it's the cas plugin that provides this API, not llvm itself.